site/vue/src/components/popup.vue

```diff
@@ -0,0 +1,119 @@
+<script setup lang="ts">
+import { watch } from 'vue';
+
+const props = withDefaults(defineProps<{
+    id?: string;
+    visible?: boolean;
+    closeButton?: {
+        id: string;
+        classes: string[];
+    };
+}>(), {
+    id: 'popup',
+    title: 'popup title',
+    closeButton() {
+        return {
+            id: 'close_button',
+            classes: ['btn-default'],
+        };
+    },
+
+});
+
+const emit = defineEmits<{
+    dismiss: [];
+}>();
+
+watch(
+    () => props.visible,
+    (visible, old_visible) => {
+        if (old_visible !== visible) {
+            if (visible) {
+                $(document).on(
+                    'keydown.popup',
+                    (event) => {
+                        if (event.key === 'Escape' && props.visible) {
+                            emit('dismiss');
+                        }
+                    });
+            }
+            else {
+                $(document).off('keydown.popup');
+            }
+        }
+    },
+    { immediate: true },
+);
+
+</script>
+
+<template>
+  <Teleport
+    v-if="visible"
+    to="body"
+  >
+    <div
+      :id="id"
+      class="popup-form"
+    >
+      <div
+        class="popup-box"
+        @click="$emit('dismiss')"
+      >
+        <div
+          class="popup-window"
+          @click.prevent
+        >
+          <slot name="title_panel">
+            <div class="form-title">
+              <slot name="title_tag">
+                <h1>
+                  <slot name="title">
+                    Untitled Form
+                  </slot>
+                </h1>
+              </slot>
+              <button
+                data-testid="close-button"
+                class="btn btn-default close-button"
+                tabindex="-1"
+                type="button"
+                @click="$emit('dismiss')"
+              >
+                Close
+              </button>
+            </div>
+          </slot>
+          <slot name="body_panel">
+            <div class="form-body">
+              <slot name="body">
+                Be sure to override the body slot.
+              </slot>
+              <slot name="buttons_panel">
```

Comment:
Is this slot necessary?  I think it would be better to force developers to put everything in a standard preformatted button bar.  If we give developers the ability to heavily customize things like this, we'll end up in the current situation with the Twig popup where every single popup is different, and it's absolutely impossible to reliably apply styling changes to all of them at once.